### ISSUE TITLE
qs: automate the construction of databases

### ISSUE THREAD

**Issue by williballenthin** [mandiant/flare-floss#796]
via #761 and @r0ny123 

> For example, to build the expert db, we can use GitHub CI, to automatically add the strings from capa rules whenever a rule with a string is added/updated in the capa rules repo.

**Comment by williballenthin** [comment]
for the #common database, this took many hours to build: a dozen hours to fetch the samples from VT, a few hours to extract strings, a few hours to index the results. im not sure this would fit within our GH Actions limits. im also not sure how frequently this data is likely to change, though its certainly worth investigating.

**Comment by williballenthin** [comment]
the #expert database is pre-populated with strings from capa rules; however, this was honestly just a shortcut to get *something* in there. we would like the #expert database to be something that is super easy for users to update and contribute back, such as with a small TUI program or github PR.

i think actually there are many bad entries in the database today from capa, things like "kernel32.dll" etc. So, im hesitate to keep pulling these strings from capa automatically. maybe we can tag update to capa-rules with followup actions to manually update the #expert database when a good string is found?

**Comment by r0ny123** [comment]
> for the #common database, this took many hours to build: a dozen hours to fetch the samples from VT, a few hours to extract strings, a few hours to index the results. im not sure this would fit within our GH Actions limits. im also not sure how frequently this data is likely to change, though its certainly worth investigating.

We can fetch that info from VT weekly/monthly basis. and regarding the GitHub action limit we can leverage some cloud platforms like AWS etc. Actually, I like the idea how [OALabs/hashdb](https://github.com/OALabs/hashdb) leveraging that.
<img width="1244" alt="209026245-1686e6fe-0130-44c7-a04e-4f7d3b77b684" src="https://github.com/mandiant/flare-floss/assets/49360849/3c9e5f81-ce13-4d55-af05-6a214f938cdc">


> maybe we can tag update to capa-rules with followup actions to manually update the #expert database when a good string is found?

This is a good idea!

**Comment by vee1e** [comment]
@mr-tz would this be in scope for the GSoC QUANTUMSTRAND? I think its a worthwhile feature to work on so that FLOSS users in the future can always trust that they're receiving the latest string tags without many false negatives.

I've done research on this but before I go all in on it I'd like your confirmation, thanks!

**Comment by mr-tz** [comment]
IMHO the capa strings are not the most relevant/highest priority db items. I'd focus first on getting library and runtime strings in a very good place and also into having very solid global prevalence data.

**Comment by mr-tz** [comment]
for reference also see https://github.com/mandiant/flare-floss/tree/quantumstrand/floss/qs/db/data/oss

**Comment by mr-tz** [comment]
@vee1e auto build from capa rules is still worth tracking/exploring in a separate issue?! Do you agree?

**Comment by vee1e** [comment]
yeah could be worth it, i'd probably want to do it before or in tandem w the msvcrt builds
